### PR TITLE
feat: Add APIGW origin for Cloudfron

### PR DESCRIPTION
**Pull request by alessio-creo**

### CODE REVIEW

File: src/main/31-cloudfront.tf [hunk 1]
```diff
@@ -16,6 +20,24 @@ module "fe_cdn" {
       domain_name           = module.fe_s3_bucket.s3_bucket_bucket_domain_name
       origin_access_control = "fe_hosting_oac"
     }
+
+    apigw = {
+      origin_id   = var.api_gateway_origin_id
+      domain_name = "${aws_api_gateway_rest_api.apigw.id}.execute-api.${var.aws_region}.amazonaws.com"
```
Comment (micdes-pagopa) [inline src/main/31-cloudfront.tf:26]: Use the`invoke_url` attribute from the APIGW stage. Note that if you use that you won't need `origin_path` here

File: src/main/31-cloudfront.tf [hunk 2]
```diff
@@ -16,6 +20,24 @@ module "fe_cdn" {
...
+      origin_path = "/${aws_api_gateway_stage.stage.stage_name}"
+
+      custom_header = [{
+        name  = "x-api-key"
+        value = aws_api_gateway_api_key.cloudfront.value
+      }]
+
+      custom_origin_config = {
+        origin_protocol_policy = "https-only"
+        origin_ssl_protocols   = ["TLSv1"]
```
Comment (micdes-pagopa) [inline src/main/31-cloudfront.tf:36]:
```suggestion
        origin_ssl_protocols   = ["TLSv1.2"]
```

File: src/main/31-cloudfront.tf [hunk 3]
```diff
@@ -16,6 +20,24 @@ module "fe_cdn" {
       domain_name           = module.fe_s3_bucket.s3_bucket_bucket_domain_name
       origin_access_control = "fe_hosting_oac"
     }
+
+    apigw = {
+      origin_id   = var.api_gateway_origin_id
```
Comment (micdes-pagopa) [inline src/main/31-cloudfront.tf:25]:
```suggestion
      origin_id   = "apigw"
```
I don't expect it to change between environments.

File: src/main/31-cloudfront.tf
Comment (micdes-pagopa) [inline src/main/31-cloudfront.tf:1]: - missing API key usage plan and association to stage
- missing CloudFront cache behaviour using the new apigw origin

**Review by micdes-pagopa — CHANGES_REQUESTED** [review]

File: src/main/31-cloudfront.tf
Comment (alessio-creo) [reply src/main/31-cloudfront.tf:1]: Dine the first. Please check if the cache behaviour is okay

**Review by micdes-pagopa — CHANGES_REQUESTED** [review]

File: src/main/31-cloudfront.tf [hunk 4]
```diff
@@ -53,21 +61,20 @@ module "fe_cdn" {
...
-    query_string    = true
+    allowed_methods          = ["GET", "HEAD", "OPTIONS"]
+    cached_methods           = ["GET", "HEAD"]
+    origin_request_policy_id = data.aws_cloudfront_origin_request_policy.all_viewer.id
+    compress                 = true
   }
   ordered_cache_behavior = [
     {
-      path_pattern           = "/*"
+      path_pattern           = "/eservices/*"
```
Comment (micdes-pagopa) [inline src/main/31-cloudfront.tf:71]: Just a note: there is a quota limit on how many cache behaviours you can have in a distribution (25).
This looks ok to me, but check with the developers how many path patterns they expect.

**Review by micdes-pagopa — CHANGES_REQUESTED** [review]

File: src/main/98-variables.tf [hunk 5]
```diff
@@ -119,6 +119,11 @@ variable "openapi_spec_path" {
   description = "The relative path in the repo for open api specification"
 }
 
+variable "api_gateway_origin_id" {
+  type        = string
+  description = "API gateway origin identification"
+}
+
```
Comment (micdes-pagopa) [inline src/main/98-variables.tf:126]:
```suggestion

```
Unused

File: src/main/env/dev/terraform.tfvars [hunk 6]
```diff
@@ -34,7 +34,8 @@ database_scaling_max_capacity = 10
 alb_ingress_group = "interop-probing-alb"
 api_version       = "v1"
 
-openapi_spec_path = "./assets/openapi_spec/interop-probing-dev-api-v1.yaml"
+openapi_spec_path     = "./assets/openapi_spec/interop-probing-dev-api-v1.yaml"
+api_gateway_origin_id = "api"
```
Comment (micdes-pagopa) [inline src/main/env/dev/terraform.tfvars:38]:
```suggestion
```
Unused

**Review by micdes-pagopa — CHANGES_REQUESTED** [review]

**Pull request merged**